app/models/concerns/wp_cache.rb

```diff
@@ -48,14 +48,21 @@ def create_or_update(wp_type, wp_id, preview = false)
     # the `update_wp_cache` instance method.
     # Removes records with unknown IDs.
     #
+    # TODO (dunyakirkali) clean up
     def create_or_update_all
-      wp_json = get_from_wp_api self.to_s.underscore.pluralize
-      ids = wp_json.map do |json|
-        wp_id = json['ID']
-        where(wp_id: wp_id).first_or_create.update_wp_cache(json)
-        wp_id
+      page = 0
+      while paginated_models.include?(wp_type) do
+        puts " page #{page}"
+        wp_json = get_from_wp_api(wp_type, page)
+        break if wp_json.empty?
+        ids = wp_json.map do |json|
+          wp_id = json['ID']
+          where(wp_id: wp_id).first_or_initialize.update_wp_cache(json)
+          wp_id
+        end
+        where('wp_id NOT IN (?)', ids).destroy_all unless ids.empty?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.